File: baekjoon/7795/main.py
t = int(input())

for i in range(t):
    
    n, m = list(map(int, input().split()))
    
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    
    a.sort()
    b.sort()
    
    bMax = max(b)
    result = 0
    
    for j in range(n):
        
        if a[j] > bMax:
            result += m
            continue
    
        l = 0
        r = m - 1
        
        ans = -1
        
        while l <= r:
            
            mid = (l + r) // 2
            
            if a[j] <= b[mid]:
                r = mid - 1
            
            else:
                l = mid + 1
                ans = mid
                
        if ans != -1:
            result += ans + 1
        
    print(result)    

File: baekjoon/7795/main.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int a[20000];
int b[20000];

int main() {

	cin.tie(0);
	ios::sync_with_stdio(false);
	
	int t;
	cin >> t;
	
	for (int i = 0; i < t; i++) {
		
		int n, m;
		cin >> n >> m;
		
		for (int j = 0; j < n; j++) {
			cin >> a[j];
		}
		
		for (int j = 0; j < m; j++) {
			cin >> b[j];
		}
		
		sort(a, a + n);
		sort(b, b + m);
		
		int bMax = b[m - 1];
		int result = 0;
		
		for (int j = n - 1; j >= 0; j--) {
			
			if (a[j] > bMax) {
				result += m;
				continue;
			}
			
			int l = 0;
			int r = m - 1;
			
			int ans = -1;
			
			// O(log M) search
			while (l < r) {
				
				int mid = (l + r) / 2;
				
				if (a[j] <= b[mid]) {
					r = mid;
				} else {
					l = mid + 1;
					ans = mid;
				}
			}
			
			if (ans != -1)
				result += ans + 1;
		}
		
		cout << result << '\n';
	}
	
	return 0;
}
